src/shared/components/download/__tests__/Download.spec.tsx

```diff
@@ -95,19 +95,23 @@ describe('Download component', () => {
     [FileFormat.tsv, true],
   ])(
     'should show column selection component when %s file type is selected and otherwise hide it',
-    (value, columnSelect) => {
+    async (value, columnSelect) => {
       const formatSelect = screen.getByTestId('file-format-select');
       fireEvent.change(formatSelect, { target: { value } });
       const customise = screen.queryByText('Customize data');
       if (columnSelect) {
-        expect(customise).toBeInTheDocument();
+        await waitFor(() => {
+          expect(customise).toBeInTheDocument();
+        });
       } else {
-        expect(customise).not.toBeInTheDocument();
+        await waitFor(() => {
+          expect(customise).not.toBeInTheDocument();
+        });
```

Comment:
This (`waitFor` + `not.toBe`) feels a bit strange. I think that if we need to wait because something is loading, we need to check for another signal which is not the thing that we're going to do the assertion on.
Either the `waitFor` is unnecessary, or we need to wait for an update of the state of the component before getting the `Customize data` (through `act`, or by `waitFor` that the selected value is the one we changed it for).

Reply:
Good catch - it wasn't needed. I got carried away with the `waitFor`s.